xbmc/games/controllers/ControllerPort.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (C) 2017-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "ControllerPort.h"
+
+#include "ControllerDefinitions.h"
+#include "utils/XMLUtils.h"
+#include "utils/log.h"
+
+#include <algorithm>
+#include <utility>
+
+using namespace KODI;
+using namespace GAME;
+
+CControllerPort::CControllerPort(std::string portId, std::vector<std::string> accepts)
+  : m_portId(std::move(portId)), m_accepts(std::move(accepts))
+{
+}
+
+void CControllerPort::Reset(void)
+{
+  CControllerPort defaultPort;
+  *this = std::move(defaultPort);
+}
+
+bool CControllerPort::IsCompatible(const std::string& controllerId) const
+{
+  return std::find(m_accepts.begin(), m_accepts.end(), controllerId) != m_accepts.end();
+}
+
+bool CControllerPort::Deserialize(const TiXmlElement* pElement)
+{
+  Reset();
```

Comment:
Should we only `Reset();` after checking `if (pElement == nullptr)`? This way you can assume that the object is unchanged when `Deserialize()` returns `false`.

Reply:
fixed